src/commonMain/kotlin/KtUniversalis.kt

```diff
@@ -14,128 +16,405 @@ import cloud.drakon.ktuniversalis.exception.InvalidParameterException
+            else -> throw UniversalisException((it.body() as ProblemDetails).toString())
+        }
+    }
+}
+
+/**
+ * Returns the data currently shown on the market board for the requested item ID and world or data center.
+ * @param worldDcRegion The world, data center, or region to retrieve data for. This may be an ID or a name. Regions should be specified as Japan, Europe, North-America, Oceania, China, or 中国.
+ * @param itemId The item ID to retrieve data for.
+ * @param listings The number of listings to return. By default, all listings will be returned.
+ * @param entries The number of recent history entries to return. By default, a maximum of `5` entries will be returned.
+ * @param noGst If the result should not have Gil sales tax (GST) factored in. GST is applied to all consumer purchases in-game, and is separate from the retainer city tax that impacts what sellers receive. By default, GST is factored in.
+ * @param hq Filter for HQ listings and entries. By default, both HQ and NQ listings and entries will be returned.
+ * @param statsWithin The amount of time before now to calculate stats over, in milliseconds. By default, this is 7 days.
+ * @param entriesWithin The amount of time before now to take entries within, in seconds. Negative values will be ignored.
+ * @param fields A list of fields that should be included in the response, if omitted will return all fields. For example if you're only interested in the listings price per unit you can set this to `listings.pricePerUnit`.
+ * @throws InvalidParameterException The parameters are invalid.
+ * @throws InvalidWorldDcItemException The world/DC or item requested is invalid.
+ * @throws UniversalisException The Universalis API returned an unexpected return code.
+ */
+suspend fun getMarketBoardCurrentData(
+    worldDcRegion: String,
+    itemId: Int,
+    listings: Int? = null,
+    entries: Int? = null,
+    noGst: Boolean? = null,
+    hq: Boolean? = null,
+    statsWithin: Int? = null,
+    entriesWithin: Int? = null,
+    fields: List<String>? = null,
+): CurrentlyShown = coroutineScope {
+    ktorClient.get("$worldDcRegion/$itemId") {
+        url {
+            if (listings != null) {
+                parameters.append("listings", listings.toString())
+            }
+            if (entries != null) {
+                parameters.append("entries", entries.toString())
+            }
+            if (noGst != null) {
+                parameters.append("noGst", noGst.toString())
+            }
+            if (hq != null) {
+                parameters.append("hq", hq.toString())
+            }
+            if (statsWithin != null) {
+                parameters.append("statsWithin", statsWithin.toString())
+            }
+            if (entriesWithin != null) {
+                parameters.append("entriesWithin", entriesWithin.toString())
+            }
+            if (fields != null) {
+                parameters.append("fields", fields.joinToString(","))
+            }
```

Comment:
I would inline those ifs to have a more condensed view.
Also doesnt parameter cast to string internally anyway?